src/harness/reference_models/aggregate_interference/aggregate_interference.py

```diff
@@ -0,0 +1,362 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+==================================================================================
+  Aggregate Interference Calculations Reference Model[R2-SGN-12].
+  Based on WINNF-TS-0112, Version V1.4.1, 16 January 2018.
+
+  This model provides the functions to calculate the aggregate interference for
+  the protection constrains of FSS, ESC, GWPZ and PPA.
+
+  The main routines are:
+
+    calculateAggregateInterferenceForFss
+    calculateAggregateInterferenceForEsc
+    calculateAggregateInterferenceForGwpz
+    calculateAggregateInterferenceForPpa
+
+  The routines returns a dictionary containing key as protection constraints 
+  (lat, long, low_freq, high_freq, entity_type) and value as the aggregate 
+  interference value in dBm for each of the constraints.
+==================================================================================
+"""
+
+import numpy as np 
+import logging
+from reference_models.interference import interference as interf
+from reference_models.geo import utils
+
+def convertAndSumInterference(cbsd_interference_list):
+  """ Converts interference in dBm to mW to calculate aggregate interference """
+  
+  interferences_in_mw = [interf.dbToLinear(cbsd_interference) 
+      for cbsd_interference in cbsd_interference_list]
+
+  aggregate_interference_in_mw = np.sum(interferences_in_mw)
```

Comment:
Not much advantage to go to np.sum, since you are converting a list to a numpy array.

Will be faster to do explicitely:
```
interf = np.array(cbsd_interference_list)
return interf.linearToDb(np.sum(interf.dbToLinear(interf)))
```
so that the dbToLinear is done in vectorized fashion

Reply:
Updated   as per the suggestion